src/Model.js

```diff
@@ -695,14 +706,16 @@ class Model {
     const attributes = filteredAttributeNames.reduce((attrs, key) => {
       const value = this[key]
       if (value) {
-        const { dataType: DataType } = attributeDefinitions[key]
+        const dataType = getDataType(value)
         let attr
-        if (DataType.name === 'Array' || DataType.name === 'Object') {
+        if (dataType === 'Array' || dataType === 'Object') {
           attr = toJS(value)
-        } else if (DataType.name === 'Date') {
+        } else if (dataType === 'Date') {
           attr = makeDate(value).toISOString()
+        } else if (dataType === 'String') {
+          attr = value.toString()
         } else {
-          attr = DataType(value)
+          if (attributeDefinitions[key].transformer) { attr = attributeDefinitions[key].transformer(value) }
```

Comment:
should the transformer be first now? previously 'Date', 'Array' were acting as a type of transformer